src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -24,21 +25,43 @@ internal abstract class AbstractTodoCommentService : ITodoCommentService
         protected abstract int GetCommentStartingIndex(string message);
         protected abstract void AppendTodoComments(ImmutableArray<TodoCommentDescriptor> commentDescriptors, SyntacticDocument document, SyntaxTrivia trivia, ArrayBuilder<TodoComment> todoList);
 
-        public async Task<ImmutableArray<TodoComment>> GetTodoCommentsAsync(
+        public async Task<ImmutableArray<TodoCommentData>> GetTodoCommentsAsync(
```

Comment:
ensure the computation of todo-commetns still happens in oop.  we may hit every file in the solution.  we do not want that to happen in proc.